python/paddle/fluid/contrib/int8_inference/utility.py

```
@@ -0,0 +1,731 @@
+        Returns:
+            The op index if the variable is the input of this op or -1 if the 
+            variable is not the input of any op. 
+        '''
+        for index, op in enumerate(program.current_block().ops[start_index:]):
+            if var_name in op.input_arg_names:
+                return index
+
+        return -1
+
+    def __get_quantize_dequantize_combination(self, program):
+        """
+        Get the quantize/dequantize op index for further inserting.
+        Args:
+            The program desc.
+        Returns:
+            Two lists contains the quantize op and dequantize op index information.
+        """
+        quantize_op_index = []
+        dequantize_op_index = []
+
+        if len(self._conv_op_index) < 2:
+            return [], []
+
+        for index, value in enumerate(self._conv_op_index):
+            if index == 0:
+                quantize_op_index.append(self._conv_op_index[index + 1] - 1)
+            elif index == len(self._conv_op_index) - 1:
+                output_var = program.current_block().ops[value].output(
+                    "Output")[0]
+                if self.__check_op_type_with_specified_var_as_input(
+                        program, output_var, index):
+                    dequantize_op_index.append(self._conv_op_index[index] + 2)
+                else:
+                    program.current_block().ops[value]._set_attr(
+                        "force_fp32_output", True)
+
+            else:
+                if self._conv_op_index[index] + 1 < self._conv_op_index[index +
+                                                                        1]:
+                    for op_index in range(self._conv_op_index[index] + 1,
+                                          self._conv_op_index[index + 1]):
+                        op_type = program.current_block().ops[op_index].type
+                        if op_type in "conv2d" and self._conv_op_index[
```

1. If conv_op_index has all the convolutions, why are you checking if there are any convolutions in between two consecutive convolutions? I see you would skip them but are there any?

---

Two consecutive convolutions means no need to insert quantize/dequantize op anymore. The code here is for safety check.

---

I see. So how about adding a comment that this is intentionally for this purpose?

---

Sure. I'll add the comments to resolve the potential confuse.

---

I thought about this again and I really suggest removal of this loop. It will never be executed based on the definition of conv_op_index. If you insist it stays, I think you could raise an exception that this should never happen. What do you think?

---

i read the snippet code several times and it looks ur comment is make sense. I'll delete those useless check and just set the force_fp32_output flag to true.